miso-web/src/main/webapp/scripts/pane_barcode.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+PaneTarget.barcode = (function() {
+  var title = "Search";
+  var url = "/miso/rest/barcodable/search";
+
+  var transform = function(searchResult) {
+    return Tile.make([
+        Tile.title(searchResult.entityType),
+        Tile.lines(["Name: " + searchResult.name, "Alias: " + searchResult.alias, "Barcode: " + searchResult.identificationBarcode])],
+      function() {
+        window.location = window.location.origin + searchResult.url;
+      }
+    );
+  };
+
+  var createHelpMessage = function() {
+    var message = document.createElement("P");
+
+    message.innerText = "Search barcodable items by name, alias, or barcode.  " +
+      "Alternatively, scan a barcode to see any matching items here";
+
+    return message;
+  };
+
+  return {
+    createPane: function(paneId) {
+      var pane = Pane.createSearchPane(paneId, title);
+      var content = pane.getElementsByClassName("widget")[0];
```

Comment:
Rather than going fishing around through the DOM, have `Pane.createSearchPane` return an object with `{content: ..., input:...}`.

Also, if all you do is update the contents of the `content` div, you could return: `{ updateContent: function(items) { updateDiv(content, items), updateTiles: function(tiles) { updateDiv.... }), input: ...}`.

Reply:
Neat!  "fishing" is "fishy".  I'll give that a try.